### PR TITLE
Move audio related code to a separate location

### PR DESCRIPTION
**Pull request by ihhub**
Decoupling the code to make it much easier to modify in the future.

### CODE REVIEW

**Review by oleg-derevenetz — APPROVED** [review]
LGTM in general, just one question regarding namespace naming.

**Comment by zenseii** [comment]
Hi @ihhub!
I get this error when running a clean build from this branch:
```16:53:54: [VERBOSE]     fheroes2::AudioInitializer::AudioInitializer:  Failed to open HEROES2X.AGG file for audio playback.```

`HEROES2X.AGG` is present in my data directory.
Its CRC32 is : `B326A02F`

**Comment by ihhub** [comment]
> Hi @ihhub! I get this error when upon booting the application after doing a clean build (i.e. wiping the build folder) from this branch: `16:53:54: [VERBOSE] fheroes2::AudioInitializer::AudioInitializer: Failed to open HEROES2X.AGG file for audio playback.`
> 
> `HEROES2X.AGG` is present in my data directory. Its CRC32 is : `B326A02F`
> 
> Background audio plays in the main menu so I take it this refers to extra object sounds/effects etc..

It was a logical mistake.

**Comment by ihhub** [comment]
Hi @oleg-derevenetz , I addressed your concern about namespace naming.

File: src/fheroes2/game/game_over.cpp [hunk 1]
```diff
@@ -326,12 +326,12 @@ fheroes2::GameMode GameOver::Result::LocalCheckGameOver()
                     res = fheroes2::GameMode::COMPLETE_CAMPAIGN_SCENARIO;
                 }
                 else {
-                    AGG::ResetAudio();
+                    AudioManager::ResetAudio();
                     Video::ShowVideo( "WIN.SMK", Video::VideoAction::WAIT_FOR_USER_INPUT );
                     // TODO : Implement function that displays the last frame of win.smk and
-                    // a dialog for name entry. AGG:PlayMusic is run here in order to start playing
+                    // a dialog for name entry. fheroes2::PlayMusic is run here in order to start playing
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/game_over.cpp:332]: I believe there should be `AudioManager::PlayMusic` now :)

**Review by oleg-derevenetz — APPROVED** [review]
Hi @ihhub I found just one minor typo in the comment, LGTM :)

File: src/fheroes2/game/game_over.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/game/game_over.cpp]: Fixed!

**Review by zenseii — APPROVED** [review]

**Pull request merged**